src/main/java/com/venafi/vcert/sdk/VCertTknClient.java

```diff
@@ -0,0 +1,247 @@
+package com.venafi.vcert.sdk;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.venafi.vcert.sdk.certificate.*;
+import com.venafi.vcert.sdk.connectors.Policy;
+import com.venafi.vcert.sdk.connectors.TokenConnector;
+import com.venafi.vcert.sdk.connectors.ZoneConfiguration;
+import com.venafi.vcert.sdk.connectors.tpp.TokenInfo;
+import com.venafi.vcert.sdk.connectors.tpp.Tpp;
+import com.venafi.vcert.sdk.connectors.tpp.TppTokenConnector;
+import com.venafi.vcert.sdk.endpoint.Authentication;
+import com.venafi.vcert.sdk.endpoint.ConnectorType;
+import feign.FeignException;
+
+import java.security.Security;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+public class VCertTknClient implements TokenConnector {
+    private static final String defaultVendorAndProductName = "Venafi VCert-Java";
+
+    private TokenConnector connector;
+
+    public VCertTknClient(Config config) throws VCertException {
+        Security.addProvider(new org.bouncycastle.jce.provider.BouncyCastleProvider());
+        switch (config.connectorType()) {
+            case TPP_TOKEN:
+                connector = new TppTokenConnector(Tpp.connect(config));
+                break;
+            default:
+                throw new VCertException("ConnectorType is not defined");
+        }
+        connector.setVendorAndProductName(isBlank(config.appInfo()) ? defaultVendorAndProductName : config.appInfo());
+    }
+
+    @VisibleForTesting
+    VCertTknClient(TokenConnector connector) {
+        this.connector = connector;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public ConnectorType getType() {
+        return connector.getType();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setBaseUrl(String url) throws VCertException {
```

Comment:
probably good to document that this is unsupported yet because now it is kinda hidden. Correct me if I am wrong but currently we have only TppTokenConnector and this will invoke its setBaseUrl which throws unsupported operation

Reply:
Added Javadoc documenting the method is not implemented and will always throw and UnsupportedOperationException.